7 create types for flight deck systems supplied data

dennis-whetten opened the pull request:
Herein is the initial proposal addressing the flight-deck provided system data issue. It is currently separated into 5 types as an initial proposal, attempting to separate out different types of data based on source and use.

dennis-whetten committed:
Added initial draft proposal of CSMIM types for avionics-sourced data

dennis-whetten committed:
Added initial draft proposal of CSMIM types for avionics-sourced data

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

dennis-whetten committed:
Updated files based on GitHub Actions Lint feedback

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

dennis-whetten committed:
Updating apState due to trailing space lint error

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

matthias-heinisch commented:
Thanks for indicating the missing attributes in resources. Fixed with #26 and rebased this PR.

uwe-salomon committed:
fix type ids, fix duration, remove default racc,
remove empty attributes, add whitespace

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

uwe-salomon commented:
I did fix the remaining issues in our automatic checks. These were purely syntactic changes, except for the "time-to-destination" resource which now is an uint in seconds; there is no "duration" type in CSMIM (yet).

uwe-salomon committed:
remove superfluent empty attributes and supertypes, fix some typos

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

matthias-heinisch commented on types/csmim.obj.aircraft.operations.1.yaml:
```diff
@@ -0,0 +1,95 @@
+
+  The aircraft operations type resides at the path "/TBD"
+
+resources:
```
should we add `utc` as an additional resource?

DennisWhetten-Boeing replied:
I was curious about the need for it and wanted to drive the discussion. The CSMIM network is required to support NTP and SNTP and I wasn't sure how CSMIM time would be handled and used vs NTP?

matthias-heinisch replied:
valid point.

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+%YAML 1.2
+---
+id: csmim.obj.aircraft.aircraft-state.1
```
Let's see if we can find an ATA chapter description for the system identifier. Maybe `information` (ATA46)?

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+    description: Bool flag indicating if the aircraft is in a state of
+      decompression
+
+  - id: doors-open
```
can we make this optional, as we don't have it on all fleet?

DennisWhetten-Boeing replied:
I think reviewing the required flag for every resource is valued. I guessed on some of them (I wanted to support this discussion at the in-person meeting) so updates from both sides may need to be reviewed.  And marking this one as optional would be fine by me if there are no objections.

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+    mode: r
+    description: Bool flag indicating if any aircraft engines are on
+
+  - id: maintenance-mode
```
do you have an overall aircraft maintenance mode?

DennisWhetten-Boeing replied:
I think most of our platforms have an airplane-wide test or maintenance mode, yes. However, this could also be used at a CSMIM-system level and managed by the CSMIM CDS. I was thinking there is value in indication to the CSMIM network that the operating mode is non-normal.

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+  - id: maintenance-mode
+    type: bool
+    mode: r
+    description: Bool flag indicating the applicable aircraft system is in a
```
what do you mean by applicable aircraft system?

DennisWhetten-Boeing replied:
In this case, by "applicable aircraft system" I meant the CSMIM CDS and associated CSMIM devices.

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+    optional: true
+    type: bool
+    mode: r
+    description: Bool flag indicating a Sat-based offboard comm system is
```
do we need to distinguish between direct air to ground, 5G, gatelink, satcom?

DennisWhetten-Boeing replied:
That is a good question. It may be more advantageous to only distinguish between high speed and low speed?  Or perhaps only a single indication for offboard connectivity availability that is true when a high speed offboard network is available?

matthias-heinisch commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,59 @@
+    description: Bool flag indicating the applicable aircraft system is in a
+      maintenance mode
+
+  - id: comm-gnd-avail
```
should we add a parameter for passenger connectivity available / cell modem status?

DennisWhetten-Boeing replied:
Per Uwe's suggestion, I split these indications into an offboard-comm type. I added comm_pax_avail for an indication of a "high-speed passenger connectivity system." I did not yet add one for cell modem; could you add more about that please?

matthias-heinisch replied:
I have updated the proposal. there is now an individual communication link status as well as this overall status. The later needs still to be linked in the path folder, my proposal is added as text in the type for now. To be discussed.

matthias-heinisch commented on types/csmim.obj.aircraft.cabin-state.1.yaml:
```diff
@@ -0,0 +1,27 @@
+  The cabin state type resides at the path "/TBD"
+
+resources:
+  - id: fsb
```
can we call it fasten-seatbelt? and can we make this optional, as not available on all our fleet?

DennisWhetten-Boeing replied:
Yes. We used "fsb" because that is the term found in some other ARINC documents (A628). Changing to fasten-seatbelt (or other clearer term) is also fine, I agree with the intent to make things more readable.

DennisWhetten-Boeing replied:
Per the following comments, perhaps the set should be: "fasten-seatbelt-active", "passenger-address-active", and "pre-recorded-audio-active"?

matthias-heinisch commented on types/csmim.obj.aircraft.cabin-state.1.yaml:
```diff
@@ -0,0 +1,27 @@
+    mode: r
+    description: Bool flag indicating fasten seat belt sign is active
+
+  - id: keyline-pa
```
maybe call ist passenger-address-active? and can we make it optional?

matthias-heinisch commented on types/csmim.obj.aircraft.cabin-state.1.yaml:
```diff
@@ -0,0 +1,27 @@
+    mode: r
+    description: Bool flag indicating if the PA Keyline is active
+
+  - id: keyline-pram
```
maybe call it pre-recorded-audio-playing? and can we make it optional?

matthias-heinisch commented on types/csmim.obj.aircraft.nav-data.1.yaml:
```diff
@@ -0,0 +1,79 @@
+%YAML 1.2
+---
+id: csmim.obj.aircraft.nav-data.1
```
can we rename nav-data to navigation? (data is already included in the topic)
also, I would split into ata sub-chapters `spatial` and `airdata` (for weather data)

DennisWhetten-Boeing replied:
Agreed, and I have made an update with a proposed split of the type to place it into the two associated topic trees. Also renamed nav-data to navigation.

matthias-heinisch commented on types/csmim.obj.aircraft.operations.1.yaml:
```diff
@@ -0,0 +1,95 @@
+%YAML 1.2
+---
+id: csmim.obj.aircraft.operations.1
```
maybe go with ata chapter naming, for example `indications` > `recorder`

matthias-heinisch commented on types/csmim.obj.aircraft.operations.1.yaml:
```diff
@@ -0,0 +1,95 @@
+    description: Phase of flight
+    enum-values:
+      - name: gate
+        key: 0
```
to be robust against empty messages (no data), should we start enumeration with `1`?

DennisWhetten-Boeing replied:
Yes that is fine, I was just copying over the phases listed in ARINC 834. Perhaps a value of 0 means either no data or invalid data (like ARINC 429 SSM)?  On that note, is there a way we can do that with bool values as well?

uwe-salomon replied:
My suggestion would be to send no message / clear the existing retained message in any of these cases. I would not create enum identifiers for failure cases, because the data topic is not meant to be used for failure detection and debugging. For bool values this is impossible anyways.

Instead, we should think about reporting detected failures (invalid sensor input etc.) via the BITE failure reporting mechanism.

matthias-heinisch reviewed (commented):
Thank you for your good contribution! I will have to discuss further internally, but find here my first thoughts.

matthias-heinisch commented on types/csmim.obj.aircraft.operations.1.yaml:
```diff
@@ -0,0 +1,95 @@
+      when a new flight leg starts
+
+  - id: flight-number
+    type: string
```
maybe an array of strings for code share flights

DennisWhetten-Boeing replied:
Agreed, change made and can be reverted if there are objections.

uwe-salomon committed:
rebase and rework to best practices and comments

- use underscore in resource names
- shorten descriptions
- make doors_open optional
- rename and make optional resources in cabin-state as per comments
- add units explicitly
- change flight phase enumeration keys

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

uwe-salomon commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,57 @@
+    mode: r
+    description: Whether any aircraft engines are on
+
+  - id: maintenance_mode
+    type: bool
+    mode: r
+    description: Whether the CSMIM system is in a maintenance mode
```
If it is CSMIM-specific, we might want to make it a separate object and place under core/csmim. If an aircraft-wide maintenance mode is not available in Airbus aircraft, they could compute the maintenance_mode indication from other parameters. In any case, I would like to know what this flag means for CSMIM systems. Are they supposed to make data load or BITE functions available only when maintenance_mode is on?

uwe-salomon commented on types/csmim.obj.aircraft.aircraft-state.1.yaml:
```diff
@@ -0,0 +1,57 @@
+    mode: r
+    description: Whether the CSMIM system is in a maintenance mode
+
+  - id: comm_gnd_avail
+    optional: true
+    type: bool
+    mode: r
+    description: Whether a terrestrial network-based offboard comm system is
+      available
+
+  - id: comm_sat_avail
+    optional: true
+    type: bool
+    mode: r
+    description: Whether a sat-based offboard comm system is available
```
Could be moved into own "communications" object. And maybe extended, as per discussion between Matthias and Dennis.

DennisWhetten-Boeing replied:
I agree with this comment and have updated the branch with a proposal.

uwe-salomon commented on types/csmim.obj.aircraft.operations.1.yaml:
```diff
@@ -0,0 +1,96 @@
+    mode: r
+    description: Phase of flight
+    enum-values:
+      - name: gate
+        key: 1
+        description: TBD
+      - name: taxi
+        key: 2
+        description: TBD
+      - name: takeoff
+        key: 3
+        description: TBD
+      - name: climb
+        key: 4
+        description: TBD
+      - name: cruise
+        key: 5
+        description: TBD
+      - name: descent
+        key: 6
+        description: TBD
```
TBD to be resolved

matthias-heinisch replied:
as discussed today, for legacy compatibility, exceptionally changed to start with 0

uwe-salomon commented on types/csmim.obj.time-triggers.1.yaml:
```diff
@@ -0,0 +1,21 @@
+  The time triggers type resides at the path "/TBD"
+
+resources:
+  - id: hr
+    type: uint
+    mode: r
+    description: The current UTC hour using 24-hour notation, updated when
+      the hour changes
+
+  - id: min
+    type: uint
+    mode: r
+    description: The current UTC minute, updated when the minute changes
```
"hour" and "minute"?

DennisWhetten-Boeing replied:
Agreed for readability. Also, should this be uint or should the whole time be sent (i.e. utc)?

uwe-salomon reviewed (commented):
Thank you for the good work so far!

dennis-whetten committed:
Updates per various comments: split nav-data into navigation and airdata, split out offboard-comm indicators, updated resource names in time-triggers.

dennis-whetten committed:
Fixed copied description in offboard-comm type.

dennis-whetten committed:
Fixed duplicate resources (not removed when type split) for offboard comm.

uwe-salomon committed:
fixed yaml error

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

matthias-heinisch committed:
split types, add path links, revise resources and units

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

matthias-heinisch committed:
yaml linting

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

matthias-heinisch committed:
fixup! split types, add path links, revise resources and units

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

dennis-whetten committed:
Updates to some types with inputs and some questions:

 csmim.obj.aircraft.flight.1
    Changed key for listed enumeration values to align with best practices:
    this was a discussion item and i think it is better to be internally
    consistent over occasionally being consistent with an external document.

    Added comments about descriptions provided.

csmim.obj.aircraft.speed.1.yaml
    Fixed inconsistency between stated units (kts) and description (former
    meters per second, now knots)

csmim.obj.aircraft.lighted_signs.1
    Changed name from lighted-signs to lighted_signs (see flight-guidance).

csmim.obj.aircraft.offboard-comm.1
    Changed enumeration to start at 1 (see above ref: aircraft.flight.1)

    Comment: Does it make sense to use enumeration here instead of a bool value
    'available'?
    Associated proposal: maybe it makes sense to use an enumeration for cabin
    announcements too instead of separate pa and pram bool resources?

csmim.obj.aircraft.orientation.1
    How is flight_path_angle different from pitch?

DennisWhetten-Boeing commented on types/csmim.obj.aircraft.orientation.1.yaml:
```diff
@@ -0,0 +1,43 @@
+    unit: °
+    description: Direction the aircraft is moving relative to true north
+
+  - id: flight_path_angle
```
What is the difference between flight_path_angle and pitch?

DennisWhetten-Boeing commented on types/csmim.obj.aircraft.offboard-comm.1.yaml:
```diff
@@ -0,0 +1,29 @@
+      "Highspeed SATCOM", "Classic SATCOM", "Airport Gatelink"
+
+resources:
+  - id: state
```
Updated state enumeration to start at 1 per best practices doc. Other question (also in commit notes): is there value in using enumeration instead of bool here?

DennisWhetten-Boeing commented on types/csmim.obj.aircraft.audio-signals.1.yaml:
```diff
@@ -0,0 +1,31 @@
+  not yet linked in path folder.
+  Requires harmonization of cabin zone layout information
+
+id: csmim.obj.aircraft.audio-signals.1
```
Does it make sense to combine resources to a single enumeration w/ states (shortened): none, pa, pram, .. etc?

matthias-heinisch replied:
OK to combine as proposed here.

DennisWhetten-Boeing commented on path/aircraft/information/cabin/connection/<name>/README.md:
```diff
@@ -0,0 +1,20 @@
+**satcom**
+satellite-based offboard communication system
+
+**satcom_geo**
```
Maybe this should be more like 'satcom_classic' for low-speed l-band satcom like SBB and 'satcom_highspeed', similar to the description in the off board-comm.1 type?

matthias-heinisch replied:
both is fine for me.

uwe-salomon committed:
create hardware part identification type

* create hardware part identification type
* add partname attribute

---------

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

dennis-whetten committed:
Updates per GitHub discussion and file notes

Split up the csmim.obj.aircraft.aircraft-state.1.yaml resources listed into
multiple types per the notes in the file, based also on the proposed paths
for each of the associated resources.
Also updated audio-signals resources from two digital discretes to a single
enumeration as a proposal. This is submitted as a discusison item, and am
happy to talk it out and/or revert it to multiple discrete (bool)  resources.
Also updated phase of flight resource in csmim.obj.aircraft.flight.1. Two main
changes: first updated the enumeration to a different ARINC standard to
align with both a definition that is already in an approved standard and
that standard also doesn't include a 0-value which aligns with the CSMIM
best practices already published, and second updated descriptions based
on the published ARINC documents for state change criteria.

dennis-whetten committed:
Updates per some comments and fixes for the linter

Lint fixes: removed trailing spaces and some duplicate empty lines.
Changes:
 Commented out all of the aircraft-state.1-.yaml file. This file is
intended to be split among multiple other type definitions.
 door-state was updated to change naming. I didn't think about it
enough and decided to combine two objectives: first most platforms
don't have a separate sensor for door ajar vs closed and latched,
and second Matthias asked about door slides. So the change is from
the is_open resource to slide_armed.
 I agree with Matthias's comment (in the commit record in GitHub)
about renaming the resource according to the agreed best practices,
so the engine-indication type resource has been changed accordingly.

dennis-whetten committed:
Linter caught issue: renamed file

matthias-heinisch committed:
create BITE reporter supertype (#32)

* inital proposal

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* remove tail from (yet unresolved) supertype reference

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* resolve some comments
- change mode to enum
- rename 2nd_obj and change to string
- rename event_codes and correct its type
- some spelling

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>
Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* use uint, rename to bite_faults, unify wording

- use uint type for unsigned integer data
- rename "bite" resource to "bite_faults"
- use the term "fault" instead of "failure" consistently

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>
Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* move failure class from object to fault dictionary (to be set per fault), select ATA6 as attribute

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* set failure class to enum, bite_data to bytes

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>

* remove mode entry, defaults to continuous monitoring

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

* add bite reporter changes as per review comments from Daniel Fischer (Airbus)

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

---------

Signed-off-by: Matthias HEINISCH <matthias.heinisch@airbus.com>
Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>
Co-authored-by: Uwe Salomon <uwe.salomon@diehl.com>

uwe-salomon committed:
create types for seats within cabin (#15)

* add seat types from seat working group

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

uwe-salomon committed:
new logo proposal (#35)

Signed-off-by: Uwe Salomon <uwe.salomon@diehl.com>

dennis-whetten committed:
Updated text and notes for airplane data proposal.

dennis-whetten committed:
Removed duplicate entries

dennis-whetten commented:
## Notes

* csmim.obj.aircraft.flight.1.yaml phase resource Similarity to A843:

        | 853 Value | 853 Name | 843 | Note |
        |-----------|----------|-----|------|
        | 1         | pre_flight_ground | GATE | 843: gnd_speed <= 2 kts |
        | 2         | taxi_out | TAXI | 843: taxi |
        | 3         | takeoff  | TAKEOFF | 843: CAS >= 80 kts |
        | 4         | climb    | CLIMB   | 843: v_speed >= 100 ft/min |
        | 5         | cruise   | CRUISE  | 843: v_speed <= 100 ft/min |
        | 6         | descent  | DESCENT | 843: v_speed <= -100 ft/min |
        | 7         | touch_down | --    | weight-on-wheels |
        | 8         | taxi_in  | TAXI    | 843: CAS <= 80 kts |
        | 9         | post_flight_ground | GATE | 843: gnd_speed <= 2 kts |


## Type to Path Mappings:

File: ../github_repo/types/csmim.obj.aircraft.audio-signals.1.yaml
   ✅ ../github_repo/path/aircraft/cabin/announcement/<location>/csmim.obj.aircraft.audio-signals.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.door-state.1.yaml
   ✅ ../github_repo/path/aircraft/cabin/lavatory/<location>/csmim.obj.aircraft.door-state.1.yaml
   ✅ ../github_repo/path/aircraft/cabin/doors/any/csmim.obj.aircraft.door-state.1.yaml
   ✅ ../github_repo/path/aircraft/cabin/doors/<location>/csmim.obj.aircraft.door-state.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.engine-indication.1.yaml
   ✅ ../github_repo/path/aircraft/indications/engine/<number>/csmim.obj.aircraft.engine-indication.1.yaml
   ✅ ../github_repo/path/aircraft/indications/engine/csmim.obj.aircraft.engine-indication.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.flight-guidance.1.yaml
   ✅ ../github_repo/path/aircraft/autoflight/guidance/csmim.obj.aircraft.flight-guidance.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.flight.1.yaml
   ✅ ../github_repo/path/aircraft/indications/recorder/flight/csmim.obj.aircraft.flight.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.id.1.yaml
   ✅ ../github_repo/path/aircraft/indications/recorder/acid/csmim.obj.aircraft.id.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.landing-gear-indication.1.yaml
   ✅ ../github_repo/path/aircraft/indications/landing-gear/csmim.obj.aircraft.landing-gear-indication.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.lighted-signs.1.yaml
   ✅ ../github_repo/path/aircraft/lights/signs/csmim.obj.aircraft.lighted-signs.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.offboard-comm.1.yaml
   ✅ ../github_repo/path/aircraft/information/cabin/connection/<name>/csmim.obj.aircraft.offboard-comm.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.orientation.1.yaml
   ✅ ../github_repo/path/aircraft/navigation/flight_env/position/orientation/csmim.obj.aircraft.orientation.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.position.1.yaml
   ✅ ../github_repo/path/aircraft/navigation/flight_env/position/csmim.obj.aircraft.position.1.yaml

File: ../github_repo/types/csmim.obj.aircraft.speed.1.yaml
   ✅ ../github_repo/path/aircraft/navigation/flight_env/speed/csmim.obj.aircraft.speed.1.yaml

File: ../github_repo/types/csmim.obj.bite-reporter.1.yaml
 ⚠️ No path found.

File: ../github_repo/types/csmim.obj.core.system-state.1.yaml
   ✅ ../github_repo/path/core/csmim/system/csmim.obj.core.system-state.1.yaml

File: ../github_repo/types/csmim.obj.directory.1.yaml
   ✅ ../github_repo/path/core/csmim/directory/csmim.obj.directory.1.yaml

File: ../github_repo/types/csmim.obj.hw-id.1.yaml
 ⚠️ No path found.

File: ../github_repo/types/csmim.obj.registration.1.yaml
   ✅ ../github_repo/path/core/csmim/registration/csmim.obj.registration.1.yaml

File: ../github_repo/types/csmim.obj.seat.1.yaml
   ✅ ../github_repo/path/airline/furnishing/seats/<row>/<seat>/csmim.obj.seat.1.yaml

File: ../github_repo/types/csmim.obj.seat.actuator.1.yaml
   ✅ ../github_repo/path/airline/furnishing/seats/<row>/<seat>/actuators/<actuator_id>/csmim.obj.seat.actuator.1.yaml

File: ../github_repo/types/csmim.obj.seat.motion.1.yaml
   ✅ ../github_repo/path/airline/furnishing/seats/<row>/<seat>/csmim.obj.seat.motion.1.yaml

File: ../github_repo/types/csmim.obj.seat.privacy-divider.1.yaml
   ✅ ../github_repo/path/airline/furnishing/seats/<row>/<seat>/dividers/<divider_id>/csmim.obj.seat.privacy-divider.1.yaml

File: ../github_repo/types/csmim.obj.time-triggers.1.yaml
   ✅ ../github_repo/path/aircraft/indications/recorder/time/utc/csmim.obj.time-triggers.1.yaml

File: ../github_repo/types/csmim.obj.weather.temperature.1.yaml
   ✅ ../github_repo/path/aircraft/navigation/flight_env/weather/temperature/csmim.obj.weather.temperature.1.yaml

File: ../github_repo/types/csmim.obj.weather.wind.1.yaml
   ✅ ../github_repo/path/aircraft/navigation/flight_env/weather/wind/csmim.obj.weather.wind.1.yaml

dennis-whetten committed:
Merge branch 'main' into 7-create-types-for-flight-deck-systems-supplied-data

Signed-off-by: dennis-whetten <dennis.whetten@boeing.com>

dennis-whetten committed:
Testing fixes for two of the github commit check errors

dennis-whetten committed:
Fixes to the commit check errors

matthias-heinisch commented on path/aircraft/cabin/announcement/<location>/csmim.obj.aircraft.audio-signals.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../../types/csmim.obj.aircraft.audio-signals.1.yaml
```
we should add a path/aircraft/cabin/announcement/<location>/README.md to list the accepted locations. Maybe add a add some generic ones, like "everywhere" or "everywhere except crew-rest" to spare message duplications?

dennis-whetten replied:
Yes, agreed. Looking through A628-P3 it specifies there are up to 10 PA Areas and channels, but we probably want more descriptive terms.

matthias-heinisch commented on types/csmim.obj.aircraft.audio-signals.1.yaml:
```diff
@@ -0,0 +1,23 @@
+    type: enum
+    mode: r
+    description: Indicates the state of the passenger address system
+    enum-values:
```
Do we need information about the source? passenger address from cockpit / purser / cabin, pre-recorded audio from in-flight entertainment / aircraft trigger ?

matthias-heinisch commented on path/aircraft/cabin/doors/<location>/csmim.obj.aircraft.door-state.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../../types/csmim.obj.aircraft.door-state.1.yaml
```
We should add a ‎path/aircraft/cabin/doors/<location>/README.md explaining the different locations passenger,cargo,service or ML1,ML2,MR1.

matthias-heinisch commented on path/aircraft/cabin/doors/any/csmim.obj.aircraft.door-state.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../../types/csmim.obj.aircraft.door-state.1.yaml
```
should we use is_open in order to have the "any"-collector with better usage - or should we have an "all" collector instead of the "any" collector?

dennis-whetten replied:
I was thinking that any vs all are two different operations of mapping many-to-one. Any is an OR operation, and all is an AND operation. So there may be different uses for both: for exterior doors we may want both available: any and all, but for others we may not care as much (e.g. engines may only care about any).
As an example: the resource for door-state is 'is_closed' (bool). So if we are saying 'all', for me that translates to 'all doors are closed', so the value changes from true to false when any door is not in the closed state. Okay, so after writing that out, it makes sense and is the use case that is probably most applicable. 
However, with engines using the 'is_running' resource, the most applicable use case is probably 'any' and not 'all'. We could always change the resource to be 'is_off' or something so the 'all' collector makes more sense for the use case?

uwe-salomon replied:
Ok, to be more precise. I'd propose that we use a collector that matches the boolean such that the combination makes sense. For doors, this would be "all" with "is_closed", or "any" with "is_open". Engines "any" with "is_running". I am fine with not using the same collector in all cases.

Alternative: Define a different type for the collection, and place it at the root, to yield something like this:

```
aircraft/cabin/doors/
  all_closed: bool
  cargo:
    closed: bool
  ML1:
    closed: bool
  ML2:
    closed: bool
```

uwe-salomon replied:
Propose to add type `csmim.obj.aircraft.doors-state.1` 
to be placed at path `aircraft/cabin/doors` (no subfolder like "any" or "all")
with the following resources:

```yaml
resources:
  - id: all_closed
    type: bool
    mode: r
  - id: all_slides_armed
    type: bool
    mode: r
```

matthias-heinisch commented on path/aircraft/indications/engine/csmim.obj.aircraft.engine-indication.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../types/csmim.obj.aircraft.engine-indication.1.yaml
```
we should place the collections consistently (i. e. having an "any" or "all" collector as for the door states)

uwe-salomon replied:
Might also be a point for the best practices?

uwe-salomon replied:
Propose to add type `csmim.obj.aircraft.engines-indication.1`
to be placed at path `aircraft/indications/engines` (**note the 's' at the end of the path**)
with the following resources:

```yaml
resources:
  - id: any_running
    type: bool
    mode: r
    description: Whether any aircraft engine is running
```

Alternative name could be "all_off".

matthias-heinisch commented on types/csmim.obj.aircraft.lighted-signs.1.yaml:
```diff
@@ -0,0 +1,23 @@
+    description: Whether the fasten seat belt sign is active
+
+  - id: no_smoking
+    type: bool
```
we could make it optional in case it becomes obsolete in future

dennis-whetten replied:
I like the idea; one can hope. However with the rise of vaping and the necessity to still make announcements about "no smoking or vaping" I think the foreseeable future still will require a lighted sign for no smoking (vaping). I think we should leave it for now until there is movement on the regulatory side, then we can update the type to make it optional?

uwe-salomon replied:
Not good to let systems depend on a resource today and optionally remove it in the future. Other way round would be ok. But this is just one boolean, which could be retain-published with a default "true" by future systems at startup. Therefore, I propose to leave it mandatory.

matthias-heinisch commented on path/aircraft/lights/signs/csmim.obj.aircraft.lighted-signs.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../types/csmim.obj.aircraft.lighted-signs.1.yaml
```
today it can be configured per zone, do we need that?

matthias-heinisch commented on path/aircraft/information/cabin/connection/<name>/csmim.obj.aircraft.offboard-comm.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../../../types/csmim.obj.aircraft.offboard-comm.1.yaml
```
can we shift this from information system (ATA46) to cabin system (ATA44-33), resulting in path/aircraft/cabin/external-communication/<name>/

dennis-whetten replied:
Sure, that makes sense.

matthias-heinisch commented on types/csmim.obj.core.system-state.1.yaml:
```diff
@@ -0,0 +1,14 @@
+%YAML 1.2
+---
+
+id: csmim.obj.core.system-state.1
```
can we decouple this from the flight deck data pull request?

uwe-salomon replied:
Sure. Should I open a PR for it?

matthias-heinisch replied:
Maybe start with an issue first. It could be part of the maintenance/fault reporting working group.

dennis-whetten replied:
I agree with that as it is maintenance / fault related.

uwe-salomon replied:
Done. See #36

uwe-salomon replied:
Please remove the type from this PR.

uwe-salomon replied:
Comment from Steve Rines:
Add aircraft maintenance mode indication.

matthias-heinisch commented on types/csmim.obj.aircraft.flight-guidance.1.yaml:
```diff
@@ -0,0 +1,20 @@
+    optional: true
+    type: uint
+    mode: r
+    unit: km
```
should we keep it in nautical miles, as we have the other values in knots?

dennis-whetten replied:
And per the discussion in the meeting on April 29, 2026; I think we should leave units as is sent over the avionics busses if possible and align across aircraft / platforms. So if platforms universally send it in nautical miles, we should state it as such.

matthias-heinisch reviewed (commented):
Thank you for your good work, we are coming close!

uwe-salomon commented on types/csmim.obj.aircraft.lighted-signs.1.yaml:
```diff
@@ -0,0 +1,23 @@
+---
+id: csmim.obj.aircraft.lighted-signs.1
+description: >
+  The lighted sings type provides discrete values, sent as booleans, indicating
```
Typo: signs

uwe-salomon commented on types/csmim.obj.time-triggers.1.yaml:
```diff
@@ -0,0 +1,20 @@
+  - id: hour
+    type: uint
+    mode: r
+    description: The current hour using 24-hour notation, updated when
```
UTC, I guess?

matthias-heinisch replied:
Note from the discussion today: let's check if it is necessary if we have NTP in the network.

dennis-whetten replied:
Well, I was thinking there were two separate requests:
One is just having the current time (e.g. ARINC 429 Label 125) sent via a parameter.
The second is having a time trigger where a periodic message is published on some time-based interval. I was assuming that this type is for the latter and not the former. The published value may be of the current hour, but the intent is to be used as a time trigger and not for a device to keep the current time based on this published value.

The question is good, and I suggest the description should be updated to clarify it is the current hour in UTC using 24-hour notation.

uwe-salomon commented on path/aircraft/cabin/lavatory/<location>/csmim.obj.aircraft.door-state.1.yaml:
```diff
@@ -0,0 +1 @@
+../../../../../types/csmim.obj.aircraft.door-state.1.yaml
```
I would leave the lavatories out of this for now.
Even though they have doors, too.

matthias-heinisch replied:
Just for clarification: The lavatory monument doors inside the cabin are not part of this door state object.
When I think about doors, I have the categories passenger doors, cargo doors and service doors in mind. The later meaning refill / refuel panels, ground power unit. There is also a lavatory service panel which could be of interest.

dennis-whetten replied:
Can you tell me more about why we should leave lab doors out of this for now? Do they have different functions that we want to expose in the base type?
I am happy to separate it out, I just am thinking about if we think about a functional aspect of a thing we are trying to characterize as a CSMIM type with data provided, then what is the difference in the data provided or the CSMIM-related functional behaviors of the type that we are trying to distinguish?

uwe-salomon replied:
I think it has to do with semantics, because these door categories seem so fundamentally different. But you are right, in the end these are doors, too, and maybe we define a lav-door type later on that derives from door-state (for semantic reasons, and to add lav-specific resources). So I withdraw my earlier comment.

Still, I would leave it out for now, until we have a lavatory group that sorts these things out in detail.

matthias-heinisch replied:
Ah, now I realize that it is linked to ATA25 here. We do not have a parameter for the lavatory monument door state. The light is directly wired in the monument, but the status is not reported as far as I know.
